source/extensions/network/dns_resolver/cares/dns_impl.h

```diff
@@ -113,5 +115,36 @@ class DnsResolverImpl : public DnsResolver, protected Logger::Loggable<Logger::I
   const absl::optional<std::string> resolvers_csv_;
 };
 
+class CaresDnsResolverFactoryImpl : public DnsResolverFactory {
+public:
+  std::string name() const override { return "envoy.dns_resolver.cares"; }
+
+  ProtobufTypes::MessagePtr createEmptyConfigProto() override {
+    return ProtobufTypes::MessagePtr{new envoy::config::core::v3::DnsResolutionConfig};
+  }
+
+  DnsResolverSharedPtr createDnsResolverCb(Event::Dispatcher& dispatcher,
```

Comment:
IIUC in this case we don't keep the same behavior because `use_tcp_for_dns_lookups` proto field isn't being read. Is this intended behavior?

Reply:
Yeah, we kept the same behavior with the latest code.